### PR TITLE
SONAR-10365 Populate pull request data

### PR DESCRIPTION
**Pull request by janos-ss**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/component/BranchPersisterImpl.java [hunk 1]
```diff
@@ -78,12 +79,25 @@ private static void checkState(boolean condition, String msg) {
...
     // MainBranchProjectUuid will be null if it's a main branch
     dto.setProjectUuid(firstNonNull(componentDto.getMainBranchProjectUuid(), componentDto.projectUuid()));
     dto.setKey(branch.getName());
     dto.setBranchType(branch.getType());
+
     // merge branch is only present if it's a short living branch
     dto.setMergeBranchUuid(branch.getMergeBranchUuid().orElse(null));
+
+    if (branch.getType() == BranchType.PULL_REQUEST) {
+      // TODO this entire object should probably come from Branch.getPullRequestData
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/component/BranchPersisterImpl.java]: Code Smell Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fjanos%2FSONAR-10365%2Fpopulate-pull-request-column&issues=AWGuwEC_aHWQ5zPscll0&open=AWGuwEC_aHWQ5zPscll0)

[ISSUE_KEY]: AWGuwEC_aHWQ5zPscll0

File: server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/component/BranchPersisterImpl.java [hunk 2]
```diff
@@ -78,12 +79,24 @@ private static void checkState(boolean condition, String msg) {
   private static BranchDto toBranchDto(ComponentDto componentDto, Branch branch) {
     BranchDto dto = new BranchDto();
     dto.setUuid(componentDto.uuid());
+
     // MainBranchProjectUuid will be null if it's a main branch
     dto.setProjectUuid(firstNonNull(componentDto.getMainBranchProjectUuid(), componentDto.projectUuid()));
+    // TODO if pull request, use PR id instead of branch name
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/component/BranchPersisterImpl.java]: Code Smell Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fjanos%2FSONAR-10365%2Fpopulate-pull-request-column&issues=AWGu8zFGaHWQ5zPscmh2&open=AWGu8zFGaHWQ5zPscmh2)

[ISSUE_KEY]: AWGu8zFGaHWQ5zPscmh2

File: server/sonar-db-dao/src/main/protobuf/db-project_branches.proto [hunk 3]
```diff
@@ -32,7 +32,7 @@ option optimize_for = SPEED;
 message PullRequestData {
   required string branch = 1;
```
Comment (teryk) [inline server/sonar-db-dao/src/main/protobuf/db-project_branches.proto]: I strongly suggest to use only optional and not use required. Recommendation from Protobuf spec. In Protobuf 3 it has been removed to only have optional elements

Comment (janos-ss) [reply server/sonar-db-dao/src/main/protobuf/db-project_branches.proto]: I didn't know, thanks. I changed to `syntax = "proto3"` now.

File: server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/component/BranchPersisterImpl.java (hunk 2 above)
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/component/BranchPersisterImpl.java]: Code Smell Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fjanos%2FSONAR-10365%2Fpopulate-pull-request-column&issues=AWGu8zFGaHWQ5zPscmh2&open=AWGu8zFGaHWQ5zPscmh2)

[ISSUE_KEY]: AWGu8zFGaHWQ5zPscmh2

Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/component/BranchPersisterImpl.java]: Code Smell Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fjanos%2FSONAR-10365%2Fpopulate-pull-request-column&issues=AWGu8zFGaHWQ5zPscmh2&open=AWGu8zFGaHWQ5zPscmh2)

[ISSUE_KEY]: AWGu8zFGaHWQ5zPscmh2

Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/component/BranchPersisterImpl.java]: Code Smell Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fjanos%2FSONAR-10365%2Fpopulate-pull-request-column&issues=AWGu8zFGaHWQ5zPscmh2&open=AWGu8zFGaHWQ5zPscmh2)

[ISSUE_KEY]: AWGu8zFGaHWQ5zPscmh2

**Pull request merged**